### PR TITLE
`@ubiquityos` gpt command

### PR DESCRIPTION
**Pull request by Keyrxng**
Resolves https://github.com/ubiquibot/plugins-wishlist/issues/29

- [worker logs](https://github.com/ubq-testing/bot-ai/issues/25#issuecomment-2226731460)

I followed your prompt template and kept the system message short and sweet.

It seems it's able to lose the question being asked so I think it might be better to prioritize the question.

I think filling the chat history slightly would do the trick

1. system
2. user - long prompt
3. assistant - manually inserted short acknowledgement of the context received
4. user - directly ask the question
5. assistant - the real API response

<hr/>

- Should this plugin be able to read it's own comments or not?
- Are we only going one level deep with the linked issue context?
- Are there to be any kind of safe guards, formatting or anything (not including the spec prompt template) included in the system message or it's free reign with little guidance like it is now?

### CODE REVIEW

File: .github/workflows/compute.yml [hunk 1]
```diff
@@ -1,4 +1,4 @@
-name: "the name of the plugin"
+name: "ubiquibot-gpt-command"
```
Comment (0x4007) [inline .github/workflows/compute.yml:1]:
```suggestion
name: "command-gpt"
```

File: src/handlers/ask-gpt.ts [hunk 2]
```diff
@@ -0,0 +1,239 @@
...
+  for (const comment of comments) {
+    const user = comment.user?.login;
+    if (
+      user === "ubiquibot" ||
+      user === "ubiquibot[bot]" ||
+      user === "ubiquibot-v2-testing" ||
+      user === "ubiquibot-dev[bot]" ||
+      user === "ubqbot[bot]" || // TODO: remove this
+      user === "github-actions[bot]"
+    ) {
```
Comment (0x4007) [inline src/handlers/ask-gpt.ts:177]: Definitely just ignore all bot comments as we do in conversation rewards.

File: src/plugin.ts [hunk 3]
```diff
@@ -1,47 +1,70 @@
...
+  } = context;
 
-  context.adapters = createAdapters(supabase, context);
+  if (isSupportedEvent(context.eventName)) {
+    const comment = context.payload.comment.body;
 
-  if (context.eventName === "issue_comment.created") {
-    // do something
+    if (!comment.startsWith("/gpt")) {
+      logger.info("Comment does not start with /gpt. Skipping.");
```
Comment (0x4007) [inline src/plugin.ts:32]: Will this be a problem? It is not clear to me if the kernel will invoke this on any new comment.

File: src/types/context.ts [hunk 4]
```diff
@@ -14,14 +14,8 @@ export interface Context<T extends SupportedEventsU = SupportedEventsU, TU exten
   eventName: T;
   payload: TU["payload"];
   octokit: InstanceType<typeof Octokit>;
-  adapters: ReturnType<typeof createAdapters>;
+  adapters: never;
```
Comment (0x4007) [inline src/types/context.ts:17]: Shouldn't this be fully removed instead of `never`?

File: src/utils/format-chat-history.ts [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+
+  const curIssue = {
+    convo: streamlined[convoKeys[0]],
+    specOrBody: specAndBodies[specAndBodyKeys[0]],
+  };
+
+  let issueSpecBlock = "";
+  issueSpecBlock += createHeader("Project Specification", specAndBodyKeys[0]);
+  issueSpecBlock += createSpecOrBody(curIssue.specOrBody);
+  issueSpecBlock += createFooter("Project Specification");
```
Comment (0x4007) [inline src/utils/format-chat-history.ts:23]: Use a `string[]` as the buffer and `.join("\n")` at the end. += is too error prone.

File: src/utils/format-chat-history.ts [hunk 6]
```diff
@@ -0,0 +1,111 @@
...
+
+export function createChatHistory(formattedChat: string) {
+  const chatHistory: ChatCompletionMessageParam[] = [];
+
+  const systemMessage: ChatCompletionMessageParam = {
+    role: "system",
+
+    content: `Using the provided context, address the question being asked and make sure to provide a clear and concise answer with no follow-up statements.
+        The LAST comment in 'Issue Conversation' is the most recent one, focus on it as that is the question being asked.
+        Use GitHub flavoured markdown in your response making effective use of lists, code blocks and other supported GitHub md features.`,
```
Comment (0x4007) [inline src/utils/format-chat-history.ts:100]: It would be useful context to explain the purpose of the plugin, and that it is being invoked inside of GitHub comments in order to help the contributors with their research.

File: src/utils/issue.ts [hunk 7]
```diff
@@ -0,0 +1,200 @@
...
+
+export async function getLinkedIssueContextFromComments(context: Context, issueComments: IssueComments) {
+  // find any linked issues in comments by parsing the comments and enforcing that the
+  // linked issue is from the same org that the current issue is from
+  const linkedIssues = await fetchLinkedIssues({ context }, issueComments);
+
+  // the conversational history of the linked issues
+  const linkedIssueComments: IssueComments = [];
+
+  // we are only going one level deep with the linked issue context fetching
```
Comment (0x4007) [inline src/utils/issue.ts:124]: Why not all? Context length is not a problem anymore.

File: src/utils/issue.ts [hunk 8]
```diff
@@ -0,0 +1,200 @@
...
+    return null;
+  }
+  if (!owner) {
+    throw new Error("Owner must be provided when parsing linked issues");
+  }
+  // the assumption here is that any special GitHub markdown formatting is converted to an anchor tag
+  const urlMatch = comment.match(/https:\/\/github.com\/([^/]+)\/([^/]+)\/(pull|issue|issues)\/(\d+)/);
+
+  /**
+   * I think we should restrict including any linked context which is not of the same org.
```
Comment (0x4007) [inline src/utils/issue.ts:144]: I think that recursion is not a problem due to generous context length for 4o

File: src/utils/issue.ts [hunk 9]
```diff
@@ -0,0 +1,200 @@
...
+   *
+   * In most cases this will be the expected behaviour, I remember a scenario where
+   * I linked to an issue in a 3rd party org, for extra reviewer context but I also include the
+   * TL;DR which is always the case. We wouldn't want that full 3rd party PR review or issue to be
```
Comment (0x4007) [inline src/utils/issue.ts:148]: Just be sure to label the section with the org/repo name and ChatGPT should know it's not as relevant

File: src/handlers/ask-gpt.ts (hunk 2 above)
Comment (Keyrxng) [reply src/handlers/ask-gpt.ts:177]: > Should this plugin be able to read it's own comments or not?

Without it seeing it's own answers it may try to re-answer a previous question and would be without relevant context. Originally I had it able to read it's own comments and had planned to do the same here

File: src/plugin.ts (hunk 3 above)
Comment (Keyrxng) [reply src/plugin.ts:32]: Truthfully me either.

The kernel does match using the regex provided in the plugin config so as far as I'm aware this probably isn't required it just felt like a good to have, I'll confirm and remove if not needed.

File: src/utils/issue.ts (hunk 7 above)
Comment (Keyrxng) [reply src/utils/issue.ts:124]: While we have a 128k window I don't think it's going to be as effective stuffing one prompt/chat history with near that amount of tokens. If we are adding more context which could potentially be 30/40k worth of tokens GPT will get lost in the sauce with it a little I think.

Also typically all issues are going to have at least one other linked issue which could easily produce deep enough recursion to max our tokens. Also that GPT output has to be factored into this 128k (likely always < 10k I'd imagine).

<hr/>

One deep seems ideal to me, it obtains strictly the most relevant context without adding too much noise but if not one level deep I'd suggest to cap it somehow whether through a depth value or token limit

File: src/utils/issue.ts (hunk 8 above)
Comment (Keyrxng) [reply src/utils/issue.ts:144]: I'm of a different mindset especially when linking to known 3rd party issues for popular packages which can often have hundreds of comments none of which are truly relevant other the what our commenter had to say before linking the issue.

Like you always say signal over noise, I don't feel it's a benefit and will cause more issues than it will solve. But I'll remove this restriction and see how it fairs.

File: src/handlers/ask-gpt.ts (hunk 2 above)
Comment (0x4007) [reply src/handlers/ask-gpt.ts:177]: I prefer to exclude for now.

File: src/plugin.ts (hunk 3 above)
Comment (0x4007) [reply src/plugin.ts:32]: @gentlementlegen might know.

File: src/utils/issue.ts (hunk 7 above)
Comment (0x4007) [reply src/utils/issue.ts:124]: I prefer to try utilizing the most amount of context. We can make assumed optimizations later.

File: src/utils/issue.ts (hunk 8 above)
Comment (0x4007) [reply src/utils/issue.ts:144]: I prefer maximizing signal to noise, this is true in the context of human communication. 
 
These LLMs on the other hand work their magic by dealing with a ton of data. I've seen better results when providing more context than less.

File: src/plugin.ts (hunk 3 above)
Comment (Keyrxng) [reply src/plugin.ts:30]: > ones like start stop which is only working for one of the two commands.

Wasn't it the case originally that the kernel was matching the command regex as opposed to `startsWith`? With regex `/(start|stop)` is technically one command so it shouldn't be treated like two as far as the kernel is concerned but the plugin would obviously have to split it.

> TL;DR it is not required but definitely safer to have the checks there.

I confirmed locally for me that it does fire on each comment so I'd say it is required or this plugin at least would comment an error for various reasons

Comment (gentlementlegen) [reply src/plugin.ts:30]: Related PR for the fix is [here](https://github.com/ubiquity/ubiquibot-kernel/issues/75).

**Comment by Keyrxng** [comment]
I'll gather fresh QA tomorrow

File: src/plugin.ts [hunk 10]
```diff
@@ -37,8 +42,8 @@ export async function plugin(inputs: PluginInputs) {
     }
 
     if (!isEnabled) {
-      logger.info("Plugin is disabled. Skipping.");
-      await addCommentToIssue(context, "The /gpt command is disabled. Enable it in the plugin settings.", true, "warning");
+      const log = logger.info("The /gpt command is disabled. Enable it in the plugin settings.");
+      await addCommentToIssue(context, log?.logMessage.diff as string);
       return;
     }
```
Comment (0x4007) [inline src/plugin.ts]: `enabled` seems like a pointless config. Please remove that.

File: src/utils/format-chat-history.ts [hunk 11]
```diff
@@ -100,7 +99,7 @@ async function createContextBlockSection(
 
   const diffBlock = [
     createHeader("Linked Pull Request Code Diff", repoString),
-    diff ? diff : "No diff available",
+    isPull ? isPull : "No diff available",
```
Comment (0x4007) [inline src/utils/format-chat-history.ts:102]: If it is not a pull shouldn't this revert or throw? Doesn't make sense that it would treat an issue as a pull.

File: tests/__mocks__/handlers.ts [hunk 12]
```diff
@@ -1,11 +1,93 @@
+/* eslint-disable @typescript-eslint/naming-convention */
+/* eslint-disable sonarjs/no-duplicate-string */
```
Comment (0x4007) [inline tests/__mocks__/handlers.ts:2]: You can recycle strings with a const in the beginning of this file.
```suggestion
```

File: tests/__mocks__/handlers.ts [hunk 13]
```diff
@@ -1,11 +1,93 @@
+/* eslint-disable @typescript-eslint/naming-convention */
```
Comment (0x4007) [inline tests/__mocks__/handlers.ts:1]: Why? Can't you just write the code to be in line with our styles?

File: tests/__mocks__/handlers.ts [hunk 14]
```diff
@@ -1,11 +1,93 @@
...
+  http.post("https://api.github.com/repos/:owner/:repo/issues", () => {
+    const id = db.issue.count() + 1;
+    const newItem = { ...issueTemplate, id };
+    db.issue.create(newItem);
+    return HttpResponse.json(newItem);
+  }),
+  // get repo issues
+  http.get("https://api.github.com/orgs/:org/repos", ({ params: { org } }: { params: { org: string } }) => {
+    return HttpResponse.json(db.repo.findMany({ where: { owner: { login: { equals: org } } } }));
+  }),
```
Comment (0x4007) [inline tests/__mocks__/handlers.ts:58]:
```suggestion
  http.get("https://api.github.com/orgs/:org/repos", ({ params: { org } }: { params: { org: string } }) => HttpResponse.json(db.repo.findMany({ where: { owner: { login: { equals: org } } } }))),
```

You can make these all one liners succienct like this.

File: tests/main.test.ts [hunk 15]
```diff
@@ -1,22 +1,323 @@
...
+  html_url: string;
+  owner: string;
+  repo: string;
+  issue_number: number;
+  issue_url?: string;
+  pull_request_url?: string;
+};
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+const octokit = jest.requireActual("@octokit/rest") as any;
```
Comment (0x4007) [inline tests/main.test.ts:34]: Why not just use the correct object? Why cast as `any`?

File: tests/main.test.ts [hunk 16]
```diff
@@ -1,22 +1,323 @@
...
+4 ubiquity: Just a comment
+=== End Linked Issue #3 Conversation ===\n
+`;
+
+    expect(infoSpy).toHaveBeenNthCalledWith(2, "Sending chat to OpenAI", {
+      caller: LOG_CALLER,
+      chat: [
+        {
+          role: "system",
+          content: `You are a GitHub integrated chatbot tasked with assisting in research and discussion on GitHub issues and pull requests.\n        Using the provided context, address the question being asked providing a clear and concise answer with no follow-up statements.\n        The LAST comment in 'Issue Conversation' is the most recent one, focus on it as that is the question being asked.\n        Use GitHub flavoured markdown in your response making effective use of lists, code blocks and other supported GitHub md features.`,
```
Comment (0x4007) [inline tests/main.test.ts:216]:
```suggestion
          content: `You are a GitHub integrated chatbot tasked with assisting in research and discussion on GitHub issues and pull requests. Using the provided context, address the question being asked providing a clear and concise answer with no follow-up statements. The LAST comment in 'Issue Conversation' is the most recent one, focus on it as that is the question being asked. Use GitHub flavoured markdown in your response making effective use of lists, code blocks and other supported GitHub md features.`,
```

Seem unnecessary.

**Comment by 0x4007** [comment]
> I'll gather fresh QA tomorrow

Curious to see examples.

File: src/plugin.ts (hunk 10 above)
Comment (Keyrxng) [reply src/plugin.ts]: Are you sure? In order to 'disable' they'd need to remove/comment out the plugin config completely from their `ubiquibot-config.yml`

I get what you mean tho either install it if you want it or don't but this way it can be configured on a per repo basis

File: src/utils/format-chat-history.ts (hunk 11 above)
Comment (Keyrxng) [reply src/utils/format-chat-history.ts:102]: just above it we return early if `!isPull` and we know it's a pull only if we can get a diff string from `octokit.pulls.get().data.diff`

File: tests/main.test.ts (hunk 15 above)
Comment (Keyrxng) [reply tests/main.test.ts]: the `jest` import from `@jest/globals` isn't the same as the namespace through `@types/jest` and types as unknown

File: tests/__mocks__/handlers.ts (hunk 13 above)
Comment (Keyrxng) [reply tests/__mocks__/handlers.ts]: Yeah it just means casting `issue_number` and `pull_number`

**Comment by Keyrxng** [comment]
@0x4007 fresh QA 

https://github.com/ubq-testing/bot-ai/issues/49#issuecomment-2247731420

**Comment by 0x4007** [comment]
Maybe let's test in production but let's make it reply to @UbiquityOS instead of /gpt

1. Make the change
2. Merge
3. Install config 

And let's go from there

**Comment by 0x4007** [comment]
Merge or delete this repo?

**Comment by Keyrxng** [comment]
Shouldn't we rebrand this repo as I think it'll be doing more than the previously intended `/ask` feature? 

- It'll respond directly to mentions as answer whatever is in the text
- I believe it's also going to be performing reviews both triggering via specific conditions and then obviously any time you want you can simply ask it to review the PR (it'll fetch it from the issue or use it from within the PR as all queries have this access)

Or should the review under conditions be it's own plugin? I assumed this would do both to save replicating the context fetching etc

**Comment by 0x4007** [comment]
Sure we can rename later

**Comment by Keyrxng** [comment]
QA: 
- https://github.com/ubq-testing/ask-plugin/issues/1

**Comment by 0x4007** [comment]
Command ask is fine for now. Your QA makes it look stable. Can we start using it?

**Comment by Keyrxng** [comment]
> command-ask is fine for now. Your QA makes it look stable. Can we start using it? Also I want to mention that I have access to o1 from the API now.

> https://platform.openai.com/docs/guides/reasoning

> I'm not sure which model is best. I'm assuming o1-mini is pretty solid for our use case though.

o1 in my opinion is too slow compared to 4o, I'd prefer to use it and honestly, reasoning models on the OpenAi website have not impressed me so far idk about you guys.

> But keep in mind o1-mini is faster, cheaper, and competitive with o1-preview at coding tasks

i.e it's faster and cheaper than o1-preview but it drags compared to 4o.

> Your QA makes it look stable. Can we start using it?

I hope so and as soon as it gets merged. I will apply the finishing touches and it should be mergeable following any other review comments.

**Comment by Keyrxng** [comment]
Typically slash command type plugins have a `commands` entry in the manifest but with this since I'm unsure what to do basically, if the command is configurable then an entry does not make sense however if it's going to be a constant then I guess I could add one.

- Currently we pass in the bot name via the config. Should this be an env var or hardcoded so partner's can't change it? Should we use the `app_slug` or the `bot.user.id` and fetch it's username?
- Since the slash command in this case now is `@UbiquityOS` which may be subject to change (if it's not subject to change then it's easy) should I write a `commands` entry or just have it forward the payload since the plugin does the processing anyway?

File: .env.example
Comment (gentlementlegen) [inline .env.example:1]: Only `.dev.vars.example` is needed isn't it?

File: .github/workflows/worker-delete.yml
Comment (gentlementlegen) [inline .github/workflows/worker-delete.yml:1]: I made little changes on `workflow-delete` and `workflow-create` sorry about that: https://github.com/ubiquity-os/plugin-template/commit/dbb9b22049cdd45bbda0230b61a5095d73771ff5

File: manifest.json [hunk 17]
```diff
@@ -0,0 +1,5 @@
+{
+  "name": "command-ask",
+  "description": "A highly context aware organization integrated chatbot",
+  "ubiquity:listeners": ["issue_comment.created"]
```
Comment (gentlementlegen) [inline manifest.json:4]: It is missing the command definition, how can it run?

File: src/handlers/ask-gpt.ts [hunk 18]
```diff
@@ -0,0 +1,47 @@
...
+}
+
+export async function askGpt(context: Context, formattedChat: string) {
+  const {
+    logger,
+    env: { OPENAI_API_KEY },
+  } = context;
+
+  if (!OPENAI_API_KEY) {
+    throw logger.error(`No OpenAI API Key detected!`);
```
Comment (gentlementlegen) [inline src/handlers/ask-gpt.ts:24]: Shouldn't this be handle when we decode the env?

File: src/types/plugin-inputs.ts [hunk 19]
```diff
@@ -18,7 +18,11 @@ export interface PluginInputs<T extends SupportedEventsU = SupportedEventsU, TU
  * The kernel will extract those and pass them to the plugin,
  * which are built into the context object from setup().
  */
-export const pluginSettingsSchema = T.Object({});
+
+export const pluginSettingsSchema = T.Object({
+  ubiquity_os_app_slug: T.String(),
```
Comment (gentlementlegen) [inline src/types/plugin-inputs.ts:23]: Should probably be camelCase

**Review by gentlementlegen — COMMENTED** [review]
Would be nice to be able to configure the ChatGpt endpoint and model through the configuration (can be inside another issue).

File: manifest.json (hunk 17 above)
Comment (Keyrxng) [reply manifest.json:4]: see [this](https://github.com/ubiquity-os-marketplace/command-ask/pull/1#issuecomment-2375667319) comment but `plugin.ts` parses the comment and skips it if the bot is not mentioned. It doesn't need an entry here to actually run it just needs the payload.

Comment (gentlementlegen) [reply manifest.json:4]: So how is that plugin used? I thought it would be something like `/ask`

Comment (Keyrxng) [reply manifest.json:4]: `@UbiquityOS ...` or `@ubiquityos ...`. Partner configurable right now, unsure if it should be hardcoded or not as we tend not to with bot names

Comment (gentlementlegen) [reply manifest.json:4]: I see. Any reason to do it that way instead of a command?

Comment (0x4007) [reply manifest.json:4]: It's more natural to use it this way

**Comment by 0x4007** [comment]
> o1 in my opinion is too slow compared to 4o

I think it's fine. A comment responding ten seconds later isn't a problem

File: .github/workflows/worker-delete.yml
Comment (Keyrxng) [reply .github/workflows/worker-delete.yml:1]: I caught those changes but thanks for the heads up

**Comment by Keyrxng** [comment]
I moved `UBIQUITY_OS_APP_SLUG` into `.env` so that we set it when we deploy the worker. I done this to make it impossible for a partner to whitelabel it and alter the command as I got the feeling that what's intended with this plugin.

File: src/types/plugin-inputs.ts [hunk 20]
```diff
@@ -18,7 +18,12 @@ export interface PluginInputs<T extends SupportedEventsU = SupportedEventsU, TU
  * The kernel will extract those and pass them to the plugin,
  * which are built into the context object from setup().
  */
-export const pluginSettingsSchema = T.Object({});
+
+export const pluginSettingsSchema = T.Object({
```
Comment (gentlementlegen) [inline src/types/plugin-inputs.ts:22]: Do we need optional with the defaults? Also be careful not sure if it works if you send an empty string to the OpenAI constructor for the base url if it works.

Comment (Keyrxng) [reply src/types/plugin-inputs.ts:22]: I referenced [conversation rewards](https://github.com/ubiquity-os-marketplace/conversation-rewards/blob/04fba4542d4cc11444f1463cd124038cddf248d6/src/configuration/content-evaluator-config.ts#L13) and it uses an empty string so I think it's safe. 

I'm not 100% sure but is the preference to not use the `T.Optional` schema?

Comment (gentlementlegen) [reply src/types/plugin-inputs.ts:22]: Because a check is done https://github.com/ubiquity-os-marketplace/conversation-rewards/blob/2da6b868c9700b68e5bbedfb64683e42b384755c/src/parser/content-evaluator-module.ts#L29

It's fine to use optional, but usually not useful if there is a default because a value will be given either way.

File: package.json [hunk 21]
```diff
@@ -79,5 +78,6 @@
     "extends": [
       "@commitlint/config-conventional"
     ]
-  }
+  },
+  "packageManager": "yarn@1.22.22"
```
Comment (0x4007) [inline package.json:82]: Why don't you downgrade to 1.22.21 so you don't have this problem anymore

File: src/helpers/format-chat-history.ts [hunk 22]
```diff
@@ -0,0 +1,161 @@
...
+
+export function createChatHistory(formattedChat: string) {
+  const chatHistory: ChatCompletionMessageParam[] = [];
+
+  const systemMessage: ChatCompletionMessageParam = {
+    role: "system",
+    content: `You are a GitHub integrated chatbot tasked with assisting in research and discussion on GitHub issues and pull requests.
+Using the provided context, address the question being asked providing a clear and concise answer with no follow-up statements.
+The LAST comment in 'Issue Conversation' is the most recent one, focus on it as that is the question being asked.
+Use GitHub flavoured markdown in your response making effective use of lists, code blocks and other supported GitHub md features.`,
```
Comment (0x4007) [inline src/helpers/format-chat-history.ts:150]: Did you test using the sandbox for optimizing the prompt? If not, please do in a new task.

File: src/types/plugin-inputs.ts [hunk 23]
```diff
@@ -18,7 +18,12 @@ export interface PluginInputs<T extends SupportedEventsU = SupportedEventsU, TU
...
+  model: T.String({ default: "o1-mini" }),
+  openAiBaseUrl: T.String({ default: "" }),
```
Comment (0x4007) [inline src/types/plugin-inputs.ts:24]: Empty string always seems wrong.

File: package.json (hunk 21 above)
Comment (Keyrxng) [reply package.json:82]: It's a feature not a problem and wasn't it agreed we'd standardize it since we are a yarn-only org with the exception of one or two bun repos? If we are no longer standardizing it I'll change my sys config

File: src/helpers/format-chat-history.ts (hunk 22 above)
Comment (Keyrxng) [reply src/helpers/format-chat-history.ts:150]: I did not, I'll extract a 50k+ token prompt and I'll do some testing with it in another task.

File: package.json (hunk 21 above)
Comment (0x4007) [reply package.json:82]: No it's not a problem for anybody except your yarn

Comment (Keyrxng) [reply package.json:82]: https://github.com/ubiquity-os-marketplace/automated-merging/pull/19#issuecomment-2345778583
https://github.com/ubiquity-os-marketplace/command-start-stop/pull/1#discussion_r1651178825
https://github.com/ubiquity/cloudflare-deploy-action/issues/2#issue-2170840592
https://github.com/ubiquity/rpc-handler/pull/39#issuecomment-2260304403

I don't understand your comment. It has been a problem multiple times and Mentlegen, Rndquu and myself have openly stated to adopt it, also you and I have had this conversation multiple times and each time previously it appeared like you had agreed so I am confused. Is it a problem or not?

Comment (0x4007) [reply package.json:82]: I never agreed

Comment (Keyrxng) [reply package.json:82]: https://github.com/ubiquity-os-marketplace/disqualifier/pull/15#discussion_r1720523533 You did here which is why I'm so confused. There are about 8 repos across the across the orgs including plugins, ubiquibot etc that have this defined. If I should remove it here should other occurrences be removed as well?

**Comment by Keyrxng** [comment]
Some recent additional QA that was built on top of this plugin:

https://github.com/ubq-testing/ask-plugin/issues/2

**Comment by Keyrxng** [comment]
This PR should be merged separate from your feature. If required branch off from this PR do not add your logic to it

This PR is held back by review only

**Comment by Keyrxng** [comment]
Realize I never pushed the branch to this repo which facilitated the onboarding bot

https://github.com/ubiquity-os-marketplace/text-vector-embeddings/pull/18
https://github.com/ubq-testing/ask-plugin/tree/onboarding-bot
